SoftLayer/CLI/vlan/create.py

```diff
@@ -0,0 +1,45 @@
+"""Order/create a VLAN instance."""
+# :license: MIT, see LICENSE for more details.
+
+import click
+from SoftLayer.managers import ordering
+
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+
+
+@click.command()
+@click.option('--name', required=False, prompt=True, help="Vlan name")
+@click.option('--datacenter', '-d', required=False, help="Datacenter shortname")
+@click.option('--pod', '-p', required=False, help="Pod name. E.g dal05.pod01")
+@click.option('--network', default='public', show_default=True, type=click.Choice(['public', 'private']),
+              help='Network vlan type')
+@click.option('--billing', default='hourly', show_default=True, type=click.Choice(['hourly', 'monthly']),
+              help="Billing rate")
+@environment.pass_env
+def cli(env, name, datacenter, pod, network, billing):
+    """Order/create a VLAN instance."""
+
+    item_package = ['PUBLIC_NETWORK_VLAN']
+    complex_type = 'SoftLayer_Container_Product_Order_Network_Vlan'
+
+    if pod and not datacenter:
+        datacenter = pod.split('.')[0]
+
+    if not network:
+        item_package = ['PRIVATE_NETWORK_VLAN']
+
+    ordering_manager = ordering.OrderingManager(env.client)
+    result = ordering_manager.place_order(package_keyname='NETWORK_VLAN',
```

Comment:
Specifying a pod should add the pod's router to the order, otherwise it will get randomly assigned to any pod in that Datacenter, which is not great for certain users.

In the container, I'm not sure which of these fields you have to set, but its one of them, I think `routerId` is the main one that will work though.

https://sldn.softlayer.com/reference/datatypes/SoftLayer_Container_Product_Order_Network_Vlan/#routerid
https://sldn.softlayer.com/reference/datatypes/SoftLayer_Container_Product_Order_Network_Vlan/#router
https://sldn.softlayer.com/reference/datatypes/SoftLayer_Container_Product_Order_Network_Vlan/#hostnamerouter